### ISSUE TITLE
[Feature Request] TorchGeo dataset support for materialized rslearn / OlmoEarth temporal data

### ISSUE THREAD

**Issue by robmarkcole** [torchgeo/torchgeo#3686]
### Summary

TorchGeo now has native temporal model support, including [ConvLSTM](https://github.com/torchgeo/torchgeo/blob/1f0e7dd10c24b0965a9352808b8e222a760dceb2/torchgeo/models/convlstm.py#L95) for inputs shaped `(B, T, C, H, W)`. At the same time, [rslearn](https://github.com/allenai/rslearn) provides strong tooling for EO data scraping, materialization, and pretraining-scale dataset creation.

A useful division of responsibility could be:

- **rslearn**: data discovery, scraping, materialization, modality assembly, preprocessing
- **TorchGeo**: dataset wrappers, datamodules, trainers, models, metrics, and training workflows

Currently, using rslearn-materialized data with TorchGeo requires custom adapter code. It would be useful to add a TorchGeo `NonGeoDataset` that can read these materialized outputs and return TorchGeo-compatible temporal samples.

### Proposal

Add an rslearn-compatible `NonGeoDataset` reader.

The initial goal is for TorchGeo to consume an existing materialized rslearn dataset layout and map it into standard TorchGeo sample dictionaries.

Proposed output sample:

```python
{
    'image': Tensor[T, C, H, W],
    'mask': Tensor[H, W] | Tensor[T, H, W],  # for segmentation
    'timestamps': Tensor[...] optional,
    'length': Tensor scalar optional,
}
```

The concrete MVP should support semantic segmentation with ConvLSTM.

### MVP Scope

Support:

- Materialized rslearn temporal imagery
- One selected input modality or configured modality stack
- Semantic segmentation target
- Output image shape `(T, C, H, W)`
- Collation to `(B, T, C, H, W)` for ConvLSTM
- Temporal padding/truncation with length, following the existing PASTIS pad_across_batches pattern
- Basic integration with [SpatioTemporalSegmentationTask](https://github.com/torchgeo/torchgeo/pull/3671)

Out of scope for MVP:

- General multi-modal foundation-model pretraining
- All OlmoEarth modality combinations
- Cloud/object-store optimizations
- Full LTAE/date-encoding support, unless timestamps are trivial to expose

### Dependency Question

Make `rslearn` an optional dependency.

If TorchGeo can read the materialized files directly without importing rslearn, that would be preferable. But if rslearn is needed to interpret dataset metadata, window definitions, normalization config, or modality schemas, then adding it as an optional dependency seems reasonable. Potential extra:

```bash
pip install "torchgeo[rslearn]"
```

### Collation

For the ConvLSTM MVP, the datamodule should reuse or generalize the existing pad_across_batches behavior used by [PASTIS](https://github.com/torchgeo/torchgeo/blob/main/torchgeo/datamodules/pastis.py)

That means variable-length temporal samples are padded or truncated along the temporal dimension to a configured padding_length, and the collated batch includes a length tensor containing the valid timestep count for each sample.

This lets ConvLSTM select the last valid timestep rather than the padded tail.

### Normalization Concern

Normalization still needs to be explicit because rslearn/OlmoEarth and TorchGeo may represent preprocessing differently. rslearn may materialize raw values, normalized values, or metadata describing normalization. TorchGeo typically applies normalization in the datamodule, not inside the dataset.

For a TorchGeo implementation, the preferred design should be:

- The dataset reads materialized values as stored.
- The matching TorchGeo datamodule owns normalization.
- If rslearn normalization metadata is available, the datamodule may optionally use it to configure TorchGeo transforms.
- The docs should clearly state whether the expected input cache is raw or already normalized.

Possible API:
```python
RslearnTemporalDataModule(
    root='...',
    normalize=True,
    normalization='rslearn',  # or 'torchgeo', 'none'
)
```

Or simpler for MVP:
```python
RslearnTemporalDataModule(
    root='...',
    normalize=True,
)
```
with documentation saying: `This assumes the materialized rslearn data is stored in raw sensor/value space. If the data has already been normalized upstream, set normalize=False.`

### Open Questions

- Is rslearn’s materialized output stable enough to support directly?
- Can TorchGeo read the output without importing rslearn, or should `rslearn` be an optional dependency?
- How should modality selection be configured?
- Should normalization be raw-by-default?
- How should timestamp metadata be exposed for future LTAE/date-aware models?
- Should this live under `torchgeo.datasets` only, or include a matching datamodule?
- What is the minimal rslearn/OlmoEarth dataset fixture needed for tests?

### Proposed MVP

Implement the smallest useful bridge:

- `RslearnTemporalDataset`
- optional `rslearn` dependency if required
- semantic segmentation samples
- temporal image output `(T, C, H, W)`
- target mask output `(H, W)`
- temporal padding collate with `length`
- example/test using ConvLSTM + `SpatioTemporalSegmentationTask`

**Comment by adamjstewart** [comment]
In light of our new AI policy, do you want to manually rewrite this issue in your own words, or should we just close it? If an issue requires scrolling through multiple pages to read, it is very unlikely to be tackled.

**Comment by robmarkcole** [comment]
closing, will create fresh issue if this becomes a requirement
